core/shared/src/main/scala/zio/internal/FiberRuntime.scala

```diff
@@ -212,8 +214,6 @@ final class FiberRuntime[E, A](fiberId: FiberId.Runtime, fiberRefs0: FiberRefs,
       }
     } finally {
       running.set(false)
-
-      if ((previousFiber ne null) || RuntimeFlags.currentFiber(_runtimeFlags)) Fiber._currentFiber.set(previousFiber)
```

Comment:
You remove this because `val previousFiber = Fiber._currentFiber.get()` was incorrect code leading to `previousFiber` being always `null`? 🤔

Reply:
Yes. Although now you mention it I'm not sure whether I should have removed the `val` instead